src/NuGet.Core/NuGet.Common/IObjectWriter.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGet.Common
+{
```

Comment:
Why this interface and its impl is part of NuGet.Common project? why not in ProjectModel itself?